apps/myjobs/app/models/workflow.rb

```diff
@@ -124,14 +124,46 @@ def stage
     Pathname.new(self.staged_dir)
   end
 
-  # Get an array of all the files of a directory
+  # Get an array of WorkflowFile Objects of all the files of a directory
   #
   # Find.find returns an enumerator - the first path is always the initial directory
   # so we return the array with the first item omitted
+  # Then, map files to WorkflowFile objects
+  #
+  # @return [WorkflowFile] An array of WorkflowFile Objects of all the files of a directory
   def folder_contents
-    File.directory?(staged_dir) ? Find.find(staged_dir).to_a[1..-1] : []
+    if File.directory?(self.staged_dir)
+      Find.find(self.staged_dir).drop(1).select {|f| File.file?(f) }.map {|f| WorkflowFile.new(f, self.staged_dir)}
+    else
+      []
+    end
   end
-
+  
+  # Return a nested array of valid files for job script field in the job options form
+  # Each file is an array with 2 elements: [relative_file_path, file_path]
+  # Relative file path to the staged dir is at index 0, which will be used as the text for the option element
+  # Full file path is at index 1, which will be used as the value for the option element
+  # 
+  # Files grouped under the same categroy are in the same array: [[relative_file_path, file_path]]
+  #
+  # Files grouped under 'Suggested files' in the dropdown are at index 1 of the array with 'Suggested files' at index 0: ["Suggested files", [[relative_file_path, file_path]]]
+  #
+  # Files grouped under 'Others' in the dropdown are at index 1 of the array with 'Others' at index 0: ["Others", [[relative_file_path, file_path]]]
+  #
+  # @return [["Suggested files",[[relative_file_path, file_path]]], ["Others",[[relative_file_path, file_path]]]] the filename string
+  def grouped_script_options
+    suggested_files = []
+    other_valid_files = []
+    folder_contents.map{ |file|
+      if file.suggested_script?
+        suggested_files.append([ file.relative_path, file.path ])
+      elsif file.valid_script?
+        other_valid_files.append([ file.relative_path, file.path ])
+      end
+    }
+    [["Suggested files", suggested_files], ["Others", other_valid_files]]
+  end
```

Comment:
Is this more readable? I don't know:

```ruby
def grouped_script_options
  {
    "Suggested files" => folder_contents.select(&:suggested_script?).map { |f| [f.relative_path, f.path] },
    "Others" => folder_contents.reject(&:suggested_script?).map { |f| [f.relative_path, f.path] },
  }.to_a
end
```

You can also keep as is which is more efficient but use `.each` here not `.map`.

Reply:
Thanks! It looks better than what I have! And I just realized `grouped_options_for_select` can take in either a hash or an array. So my new commit doesn't have `to_a`.